webmin/os-eol-lib.pl

```diff
@@ -0,0 +1,230 @@
+                        if ($ago->{"${unit}s"}) {
+                                my $value = abs($ago->{"${unit}s"});
+                                return $value == 1 ? "1 " . $text{"os_eol_${unit}"} :
+                                        "$value " . $text{"os_eol_${unit}s"};
+                                }
+                        }
+                };
+        $eol_data->{'_eol'} = $eol_date->{'short'};
+        $eol_data->{'_eol_in'} = $eol_in->($eol_date);
+        if ($eol_data->{'extendedSupport'}) {
+                my $eol_date = &make_date(
+                        $eol_data->{'_ext_eol_timestamp'}, { '_' => 1 });
+                $eol_data->{'_ext_eol'} = $eol_date->{'short'};
+                $eol_data->{'_ext_eol_in'} = $eol_in->($eol_date);
+                }
+        }
+else {
+        $eol_data->{'_eol'} =
+                &make_date($eol_data->{'_eol_timestamp'}, 1);
+        if ($eol_data->{'extendedSupport'}) {
+                $eol_data->{'_ext_eol'} =
+                        &make_date($eol_data->{'_ext_eol_timestamp'}, 1);
+                }
+        }
+
+# Is expired?
+my $expired = ($eol_data->{'_ext_eol_timestamp'} || 
+               $eol_data->{'_eol_timestamp'}) < time();
+$eol_data->{'_expired'} = $text{'os_eol_reached'} if ($expired);
+
+# Is expiring (in 3 months by default, unless configured otherwise)
+my $os_eol_before = $gconfig{'os_eol_before'} // 3;
+if (!$expired && $os_eol_before) {
+        my $expiring = ($eol_data->{'_ext_eol_timestamp'} ||
+                        $eol_data->{'_eol_timestamp'}) < time() +
+                                60*60*24*30*$os_eol_before ? 1 : 0;
+        # Set the expiring message
+        if ($expiring) {
+                $eol_data->{'_expiring'} =
+                        $eol_data->{'_eol_in'} ?
+                                $text{'os_eol_reaching'} . " " .
+                                        $eol_data->{'_eol_in'} :
+                                $text{'os_eol_reaching2'};
+                }
+        # Set extended support expiring message (if available)
+        if ($eol_data->{'extendedSupport'}) {
+                my $expiring = $eol_data->{'_ext_eol_timestamp'} < time() +
+                                60*60*24*30*$os_eol_before ? 1 : 0;
+                if ($expiring) {
+                        $eol_data->{'_ext_expiring'} =
+                                $eol_data->{'_ext_eol_in'} ?
+                                        $text{'os_eol_reaching'} . " " .
+                                                $eol_data->{'_ext_eol_in'} :
+                                        $text{'os_eol_reaching2'};
+                        }
+                }
+        }
+return $eol_data;
+}
+
+# eol_update_cache()
+# Caches the use of EOL data for next 30 days
+# or until the next Webmin update
+sub eol_update_cache
+{
+my %miniserv;
+&get_miniserv_config(\%miniserv);
+if ($gconfig{'os_eol_db'} ne $miniserv{'server'}) {
```

Comment:
What does the `miniserv{'server'}` field contain?

Reply:
Current Webmin version.

Reply:
Though it also contains `Miniserv/` string but that's not important as long as it's different.

Reply:
Oh .. better to use the get_webmin_version function then

Reply:
That's really best-effort ... we have a function to get the version, so we should just use it